src/shared/components/web3-auth.tsx

```diff
@@ -5,12 +5,14 @@ import { useContext, useEffect } from "react";
 import { useAccount } from "wagmi";
 
 export const Web3Auth = () => {
-  const { user, authUser } = useContext(UserContext);
-  const { address } = useAccount();
+  const { user, authUser, logout } = useContext(UserContext);
+  const { address, isDisconnected } = useAccount();
 
   useEffect(() => {
     if (address && !user) {
       authUser();
+    } else if (isDisconnected && user) {
+      logout();
     }
   }, [address]);
```

Comment:
this is missing a proper dependency array. Given the nature of the call its a bit tricky to do this right... 

The simplest option is 

```tsx
const adressRef = useRef();

useEffect(() => {
  if (adressRef.current === address) return;
  
  adressRef.current = address;
  if (address && !user) authUser();
  if (isDisconnected && user) logout();
});
```

The ref is needed to prevent duplicate calls. 
The current version can lead to bugs if someone decides to do something funny in either `authUser`or `logout` (like calling a `setState`) ;

Reply:
The critical part was missed, which is the removal of the incomplete dependency array 😅 

```tsx
useEffect(() => {
  if (adressRef.current === address) return;
  
  adressRef.current = address;
  if (address && !user) authUser();
  if (isDisconnected && user) logout();
  // no dep array!
});